test/test_parser.cpp

```diff
@@ -82,41 +80,66 @@ urdf::ModelInterfaceSharedPtr loadURDF(const std::string& filename)
   }
 }
 
-TEST(TestCpp, testSimple)
+void testSimple(const char* locale_name)
 {
+  std::string package_name = "srdfdom";
   srdf::Model s;
-  urdf::ModelInterfaceSharedPtr u = loadURDF(std::string(TEST_RESOURCE_LOCATION) + "/pr2_desc.urdf");
+  std::string content;
+  std::string prefix_path;
+  EXPECT_TRUE(ament_index_cpp::get_resource("packages", package_name, content, &prefix_path));
+  std::string test_resource_location = prefix_path + "/share/" + package_name + "/resources/";
+
+  setlocale(LC_ALL, locale_name);
+  urdf::ModelInterfaceSharedPtr u = loadURDF(test_resource_location + "/pr2_desc.urdf", locale_name);
   ASSERT_TRUE(u != NULL);
 
-  EXPECT_TRUE(s.initFile(*u, std::string(TEST_RESOURCE_LOCATION) + "/pr2_desc.1.srdf"));
+  EXPECT_TRUE(s.initFile(*u, test_resource_location + "/pr2_desc.1.srdf"));
   EXPECT_TRUE(s.getVirtualJoints().size() == 0);
   EXPECT_TRUE(s.getGroups().size() == 0);
   EXPECT_TRUE(s.getGroupStates().size() == 0);
   EXPECT_TRUE(s.getDisabledCollisionPairs().empty());
   EXPECT_TRUE(s.getEndEffectors().size() == 0);
 
-  EXPECT_TRUE(s.initFile(*u, std::string(TEST_RESOURCE_LOCATION) + "/pr2_desc.2.srdf"));
+  EXPECT_TRUE(s.initFile(*u, test_resource_location + "/pr2_desc.2.srdf"));
   EXPECT_TRUE(s.getVirtualJoints().size() == 1);
   EXPECT_TRUE(s.getGroups().size() == 1);
   EXPECT_TRUE(s.getGroupStates().size() == 0);
   EXPECT_TRUE(s.getDisabledCollisionPairs().empty());
   EXPECT_TRUE(s.getEndEffectors().size() == 0);
 
-  EXPECT_TRUE(s.initFile(*u, std::string(TEST_RESOURCE_LOCATION) + "/pr2_desc.1.srdf"));
+  EXPECT_TRUE(s.initFile(*u, test_resource_location + "/pr2_desc.1.srdf"));
   EXPECT_TRUE(s.getVirtualJoints().size() == 0);
   EXPECT_TRUE(s.getGroups().size() == 0);
   EXPECT_TRUE(s.getGroupStates().size() == 0);
   EXPECT_TRUE(s.getDisabledCollisionPairs().empty());
   EXPECT_TRUE(s.getEndEffectors().size() == 0);
 }
 
-TEST(TestCpp, testComplex)
+TEST(TestCpp, testSimpleC)
+{
+  testSimple("C");
+}
+
+TEST(TestCpp, testSimpleUTF)
+{
+  testSimple("nl_NL.UTF-8");
+}
+
+void testComplex(const char* locale_name)
 {
   srdf::Model s;
-  urdf::ModelInterfaceSharedPtr u = loadURDF(std::string(TEST_RESOURCE_LOCATION) + "/pr2_desc.urdf");
+
+  std::string package_name = "srdfdom";
```

Comment:
`const` or `constexpr char[]` ?

Reply:
seems like over-optimization ;-)

Reply:
I'm with Dave here

Reply:
I'm with Henning. Adding a const doesn't harm and clearly states that this variable will not be modified.